src/lib/device-config.ts

```diff
@@ -42,9 +44,10 @@ export const generateConfig = (
 	const userPromise = getUser(req);
 
 	// Devices running ResinOS >=1.2.1 are capable of using Registry v2, while earlier ones must use v1
-	const registryHost = resinSemver.gte(osVersion, '1.2.1')
-		? REGISTRY2_HOST
-		: REGISTRY_HOST;
+	if (resinSemver.lte(osVersion, '1.2.0')) {
+		throw new BadRequestError('Deprecated ResinOS version');
```

Comment:
`resin` should be `balena` and specifying the version range makes it easier for users to fix the error :)
```suggestion
		throw new BadRequestError('balenaOS versions <= 1.2.0 are no longer supported, please update');
```